Engine/DataFeeds/Enumerators/Factories/FuturesChainUniverseSubscriptionEnumeratorFactory.cs

```diff
@@ -0,0 +1,123 @@
+        {
+            _isLiveMode = false;
+            _enumeratorConfigurator = enumeratorConfigurator;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FuturesChainUniverseSubscriptionEnumeratorFactory"/> class
+        /// </summary>
+        /// <param name="symbolUniverse">Symbol universe provider of the data queue</param>
+        public FuturesChainUniverseSubscriptionEnumeratorFactory(IDataQueueUniverseProvider symbolUniverse, ITimeProvider timeProvider)
+        {
+            _isLiveMode = true;
+            _symbolUniverse = symbolUniverse;
+            _timeProvider = timeProvider;
+            _enumeratorConfigurator = (sr, input) => input;
+        }
+
+        /// <summary>
+        /// Creates an enumerator to read the specified request
+        /// </summary>
+        /// <param name="request">The subscription request to be read</param>
+        /// <param name="dataFileProvider">Provider used to get data when it is not present on disk</param>
+        /// <returns>An enumerator reading the subscription request</returns>
+        public IEnumerator<BaseData> CreateEnumerator(SubscriptionRequest request, IDataFileProvider dataFileProvider)
+        {
+            if (_isLiveMode)
+            {
+                var localTime = request.StartTimeUtc.ConvertFromUtc(request.Configuration.ExchangeTimeZone);
+                
+                // loading the list of futures contracts and converting them into zip entries
+                var symbols = _symbolUniverse.LookupSymbols(request.Security.Symbol.Underlying.Value, request.Security.Type);
+                var zipEntries = symbols.Select(x => new ZipEntryName { Time = localTime, Symbol = x } as BaseData).ToList();
+
+                var underlyingEnumerator = new TradeBarBuilderEnumerator(request.Configuration.Increment, request.Security.Exchange.TimeZone, _timeProvider);
+                underlyingEnumerator.ProcessData(new Tick { Value = 0 });
+
+                // configuring the enumerator
+                var subscriptionConfiguration = GetSubscriptionConfigurations(request).First();
+                var subscriptionRequest = new SubscriptionRequest(request, configuration: subscriptionConfiguration);
+                var configuredEnumerator = _enumeratorConfigurator(subscriptionRequest, underlyingEnumerator);
+
+                return new DataQueueFuturesChainUniverseDataCollectionEnumerator(request.Security.Symbol, configuredEnumerator, zipEntries);
+            }
+            else
+            {
+                var factory = new BaseDataSubscriptionEnumeratorFactory();
+
+                var enumerators = GetSubscriptionConfigurations(request)
+                    .Select(c => new SubscriptionRequest(request, configuration: c))
+                    .Select(sr => _enumeratorConfigurator(request, factory.CreateEnumerator(sr, dataFileProvider)));
```

Comment:
If this enumerator is use very often, it will be more performant to combine these two `Select` statements into 1.

Each call to `Select` generates an enumerator state machine which can cause undue overhead if not needed.